hw/ip/otbn/doc/_index.md

```diff
@@ -490,6 +490,35 @@ The error that caused the alert can be determined by reading the {{< regref "FAT
 If OTBN was running, this value will also be reflected in the {{< regref "ERR_BITS" >}} register.
 A fatal alert can only be cleared by resetting OTBN through the `rst_ni` line.
 
+
+### Error Prioritization
+
+OTBN can signal multiple errors at once, by setting multiple bits of {{< regref "ERR_BITS" >}}.
+There are some scenarios where one error takes priority over another.
+In these scenarios the higher priority error will suppress some other errors.
+
+* Any error that produces a fatal alert will suppress all non fatal errors.
+* Any `call_stack` error that occurs due to a pop will suppress:
+  - `bad_data_addr`
+  - `bad_insn_addr`
+  - `loop`
```

Comment:
Hmm, I'm a bit confused. I assumed that the RTL that's in charge of going "hey, there's a loop instruction in a bad place" wouldn't depend on the register value that got popped. So I'm not really sure why this is so difficult to implement. (Probably just me not understanding something!)

Maybe it would make sense to try implementing the "most obvious" semantics first and then only make things more complicated in the spec if that's really gnarly? If we do that, we can at least explain why we have our weird corner cases.

Reply:
It's not especially complex, just adds a bit of extra wiring between the controller and the loop_controller.

However I think the presence of 'call_stack' suppressing any 'loop' error is "most obvious" semantics. As explained in the other comment you say the pop occurs first and only if that is successful does anything else happen.

Otherwise you're saying if a call stack pop fails that suppresses errors that depend directly upon the result of that 'call_stack' pop but other aspects of the instruction that don't depend upon the call stack could still cause errors. In particular for `loop x1, 2` you could see a `call_stack` and `loop` error where that `loop x1, 2` is the final instruction of another loop or if it's pushing to a full loop stack.

I felt the suppression here gives simpler RTL as well as simpler architecture.